### PR TITLE
Add ESP32-S3 SPI-Flash support

### PR DESCRIPTION
**Pull request by Ouss4**
## Summary
Add support for the SPI-Flash for ESP32-S3.
## Impact
New feature -- no impact on existing code.
## Testing
Provided defconfig.

### CODE REVIEW

**Review by gustavonihei — APPROVED** [review]
LGTM

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c [hunk 1]
```diff
@@ -0,0 +1,807 @@
...
+            .erase  = esp32s3_erase,
+            .bread  = esp32s3_bread_decrypt,
+            .bwrite = esp32s3_bwrite_encrypt,
+            .read   = esp32s3_read_decrypt,
+            .ioctl  = esp32s3_ioctl,
+#ifdef CONFIG_MTD_BYTE_WRITE
+            .write  = NULL,
+#endif
+            .name   = "esp32s3_spiflash_encrypt"
+          }
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c:142]:
```suggestion
          },
```
This missing comma is breaking the build.

File: arch/xtensa/src/esp32s3/esp32s3_spiflash.c [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+static void spiflash_start(void);
+static void spiflash_end(void);
+
+/****************************************************************************
+ * Public Functions Declaration
+ ****************************************************************************/
+
+extern void spi_flash_guard_set(const struct spiflash_guard_funcs_s *funcs);
+extern uint32_t cache_suspend_icache(void);
+extern void cache_resume_icache(uint32_t val);
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32s3/esp32s3_spiflash.c:140]:
```suggestion
extern void spi_flash_guard_set(const struct spiflash_guard_funcs_s *funcs);
extern int cache_invalidate_addr(uint32_t addr, uint32_t size);
extern uint32_t cache_suspend_icache(void);
extern void cache_resume_icache(uint32_t val);
```
Build is failing on a warning for missing declaration of `cache_invalidate_addr`.

File: arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c (hunk 1 above)
Comment (Ouss4) [reply arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c]: Thanks! Fixed!

File: arch/xtensa/src/esp32s3/esp32s3_spiflash.c (hunk 2 above)
Comment (Ouss4) [reply arch/xtensa/src/esp32s3/esp32s3_spiflash.c:140]: Thanks! Fixed!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**